pandas_gbq/gbq.py

```diff
@@ -1007,6 +1007,12 @@ def to_gbq(
         raise NotFoundException(
             "Invalid Table Name. Should be of the form 'datasetId.tableId' "
         )
+    has_credentials = (
+        credentials is not None
+        and credentials.__class__.__name__.endswith("Credentials")
+    )
+    if project_id is None and has_credentials:
+        project_id = credentials.project_id
```

Comment:
As far as I know, only ServiceAccount credentials actually have a [`project_id`](https://google-auth.readthedocs.io/en/latest/reference/google.oauth2.service_account.html#google.oauth2.service_account.Credentials.project_id) property.

I'd actually much prefer that we do the following condition:

```
if project_id is None and hasattr(credentials, "project_id"):
```

and forget `has_credentials` altogether.

Reply:
That's much cleaner indeed